packages/relay-runtime/network/__tests__/RelayObservable-test.js

```diff
@@ -1444,6 +1444,28 @@ describe('RelayObservable', () => {
       expect(list).toEqual(['error', error]);
     });
 
+    it('Converts an AsyncGenerator to an Observable', done => {
+      const value = {key: 'value'};
+
+      async function* generate() {
+        yield value;
+      }
+      const result = generate();
+      const obs = RelayObservable.from(result);
+
+      obs.subscribe({
+        next: val => {
+          expect(val).toEqual(value);
+        },
+        error: err => {
+          done(err);
+        },
+        complete: () => {
+          done();
+        },
+      });
+    });
+
```

Comment:
let's add a test for error handling and for cleanup

Reply:
I think cleanup is very tricky to unit test, i have "tested' this manually, but is very hard to actually test because you cannot step through the yields. 

One theory i had was to:

```js
async function* generate() {
    yield 'a';
    yield 'b';
}
```

but there is no way to expect a value, then unsubscribe. 

Calling .unsubscribe doesn't run `complete` (which is probably a bug). But that aside, if you know of a better way to test this, please let me know! Im stuck for ideas.